realm/realm-library/src/main/java/io/realm/RealmObject.java

```diff
@@ -150,25 +150,86 @@ public final boolean isValid() {
     }
 
     /**
-     * Determines if the current RealmObject is obtained synchronously or asynchronously (from a worker thread).
+     * Checks if the query used to find this RealmObject has completed.
+     *
+     * Objects created from async methods like {@link RealmQuery#findFirstAsync()} return a
+     * dummy object that represents the future result of the {@link RealmQuery}. It can be considered
+     * similar to a {@link java.util.concurrent.Future} in this regard.
+     *
+     * Once this method returns {@code true}, this object represents the query result even if the query
+     * didn't find any object matching the query parameters. In this case the {@link RealmObject} will
+     * become a "null" object.
+     *
+     * "Null" objects represents {@code null} and will throw if any accessor is called, so it is important
```

Comment:
"will throw": the object doesn't throw. Maybe:

"Null" objects represent {@code null}. An exception is throw if any accessor is called ..."